source/adapters/opencl/memory.cpp

```diff
@@ -224,13 +226,13 @@ cl_map_flags convertURMemFlagsToCL(ur_mem_flags_t URFlags) {
 UR_APIEXPORT ur_result_t UR_APICALL urMemBufferCreate(
     ur_context_handle_t hContext, ur_mem_flags_t flags, size_t size,
     const ur_buffer_properties_t *pProperties, ur_mem_handle_t *phBuffer) {
-
   cl_int RetErr = CL_INVALID_OPERATION;
+  UR_RETURN_ON_FAILURE(urContextRetain(hContext));
```

Comment:
This is temporary on this PR to pass the intel/llvm tests, but will be fixed in this [PR](https://github.com/oneapi-src/unified-runtime/pull/1230), so after [1230](https://github.com/oneapi-src/unified-runtime/pull/1230) is merged, this PR will be rebased on top of it and be ready to merge.